### PR TITLE
Python Benchmarks

### PR DESCRIPTION
**Pull request by chaudum**
@mfussenegger  👯

### CODE REVIEW

File: api/src/crate/benchapi/application.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    Resource for doc.benchmarks
+    Supported method: GET
+    """
+
+    def get(self, group):
+        param_from = request.args.get('from')
+        param_to = request.args.get('to')
+        params = []
+        where_clause = []
+        time_from = datetime.now()
```
Comment (mfussenegger) [inline api/src/crate/benchapi/application.py]: probably doesn't matter because the time is stripped anway - but I think the build hash time in Crate is utc - so might make sense to use `utcnow` here.

File: py/specs/insert_bulk.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        statements=[
+            "create table t_bulk_insert (id integer, value string) with (number_of_replicas=0)",
+        ]
+    ),
+    teardown=Instructions(
+        statements=[
+            "drop table t_bulk_insert",
+        ]
+    ),
+    queries=[
```
Comment (mfussenegger) [inline py/specs/insert_bulk.py]: Just curious: Why didn't you use ``[[load_data]]`` here? The intention for `load_data` was to be able to benchmark bulk inserts =)

File: py/specs/sys_shards.toml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+[setup]
+statements = [
+    "create table t_shards (id int, name string, value float) clustered into 1000 shards with (number_of_replicas=0)",
+    "insert into t_shards (id, name, value) values (1, 'foo', 0.5), (2, 'bar', 1.0), (3, 'foobar', 1.5), (4, 'baz', 2.0)",
```
Comment (mfussenegger) [inline py/specs/sys_shards.toml]: how is this insert relevant?

Comment (chaudum) [inline py/specs/sys_shards.toml]: not

File: api/src/crate/benchapi/application.py (hunk 1 above)
Comment (chaudum) [inline api/src/crate/benchapi/application.py]: 👍🏼

File: api/src/crate/benchapi/application.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        if group and group in mapping:
+            params.append(group)
+            where_clause.append("WHERE meta['name'] = ANY(?)")
+            params.append(mapping[group])
+        else:
+            return self.error('No or invalid benchmark group specified')
+
+        if param_from:
+            try:
+                time_from = datetime.fromtimestamp(mktime(strptime(param_from, '%Y-%m-%d')))
```
Comment (chaudum) [inline api/src/crate/benchapi/application.py]: it is

**Comment by mfussenegger** [comment]
lgtm
